lib/spack/spack/test/spec_semantics.py

```diff
@@ -1004,6 +1006,29 @@ def test_spec_override(self):
         assert new_spec.compiler_flags["cxxflags"] == ["-O1"]
 
 
+@pytest.mark.parametrize(
+    "spec_str,format_str,expected",
+    [
+        ("zlib@git.foo/bar", "{name}-{version}", str(pathlib.Path("zlib-git.foo_bar"))),
+        ("zlib@git.foo/bar", "{name}/{version}", str(pathlib.Path("zlib", "git.foo_bar"))),
+        ("zlib@git.foo/bar", r"C:\\installroot\{name}\{version}", None),
+        (
+            "zlib@{0}=1.0%gcc".format("a" * 40),
```

Comment:
```suggestion
            "zlib@git.features/new-feature=1.0%gcc",
```
We should be testing a case with a `/` in the version, in addition to testing for `=`.

Reply:
Separate test added